src/main/java/com/Novi/TechItEasy/services/TelevisionService.java

```diff
@@ -145,6 +151,85 @@ public TelevisionDto changeTelevision(Long id, TelevisionInputDto television) {
     }
 
 
+    ///// For assigning remote controllers to television by id /////
+    public TelevisionDto assignRemoteControllerToTelevision(Long televisionId, IdInputDto remoteId) {
+        Optional<Television> fetchedTelevision =  televisionRepository.findById(televisionId);
+        Optional<RemoteController> fetchedRemoteController = remoteControllerRepository.findById(remoteId.id);
+
+        if (fetchedTelevision.isPresent()) {
+            if (fetchedRemoteController.isPresent()) {
+
+                fetchedTelevision.get().setRemoteController(fetchedRemoteController.get());
+
+                televisionRepository.save(fetchedTelevision.get());
+                fetchedRemoteController.get().setTelevision(fetchedTelevision.get());
+
+                return TelevisionDto.fromTelevision(televisionRepository.findById(televisionId).get());
```

Comment:
Het is niet de bedoeling om meerdere keren de .get() methode van een Optional aan te roepen als je dat kunt voorkomen, maar dat vraagt nog niet heel veel extra rekenkracht. Het dubbel aanroepen van de database vraagt beduidend meer rekenkracht en wil je zo veel mogelijk vermijden.